test/edu/ucsb/nceas/metacat/dataone/D1AuthHelperTest.java

```diff
@@ -157,44 +200,270 @@ public void setUp() throws Exception {
         cn1CNSession = new Session();
         cn1CNSession.setSubject(TypeFactory.buildSubject("cn1Subject"));
 
+        // Create a session with a null subject
+        nullSession = new Session();
+        nullSession.setSubject(TypeFactory.buildSubject(null));
+
+        // Create a session without setting the subject
+        missingSubjectSession = new Session();
+
+        // Create a session with an empty subject value
+        emptySubjectSession = new Session();
+        emptySubjectSession.setSubject(TypeFactory.buildSubject(""));
+
+        // Create Metacat Admin session
+        metacatAdminSession = new Session();
+        metacatAdminSession.setSubject(TypeFactory.buildSubject("http://orcid.org/0000-0002-6076-8092"));
+
+        // Create a second Metacat Admin session
+        metacatAdminOtherSession = new Session();
+        metacatAdminOtherSession.setSubject(TypeFactory.buildSubject("http://orcid.org/0000-0003-0077-4738"));
+
+        // Create a Local Node session
+        localNodeSession = new Session();
+        localNodeSession.setSubject(TypeFactory.buildSubject("CN=urn:node:METACAT1,DC=dataone,DC=org"));
+
+        // Create an unauthorized session
+        notAuthorizedSession = new Session();
+        notAuthorizedSession.setSubject(TypeFactory.buildSubject("notAFriend"));
     }
 
-    // TODO: This test should be implemented when time permitting.
-    @Ignore("Not yet implemented...")
+    /**
+     * Check expandRightsHolder returns true with approved subject, ex. the Session subject is a
+     * member of the rights holder group.
+     */
     @Test
-    public void testExpandRightsHolder() {
-        fail("Not yet implemented");
+    public void testExpandRightsHolder() throws Exception {
+        // expandRightsHolder() is a public static method
+        // D1Client contains methods that make network calls, which is to be mocked
+        try (MockedStatic<D1Client> mockD1Client = Mockito.mockStatic(D1Client.class)) {
+            // Get a default rightsHolderGroup list
+            List<Group> testRightsHolderGroup = getGenericRightsHolderGroup(null);
+
+            SubjectInfo mockSInfo = Mockito.mock(SubjectInfo.class);
+            when(mockSInfo.getGroupList()).thenReturn(testRightsHolderGroup);
+
+            CNode mockCN = Mockito.mock(CNode.class);
+            // .listSubjects(...) makes a network call
+            when(mockCN.listSubjects(eq(null), any(), eq(null), anyInt(), anyInt()))
+                .thenReturn(mockSInfo);
```

Comment:
why can't we move these lines inside the private method, too? They are repeated in all 4 tests

Reply:
I left them out as I had wanted the tests to show more of the mocking - but you're right, they are repeated unnecessarily (and we can always look at the new method `getMockCN` to see how the mocking works). This has been refactored, thank you!